pkg/agent/manager/sync.go

```diff
@@ -36,6 +36,18 @@ type Cache interface {
 	GetStaleEntries() []*cache.StaleEntry
 }
 
+func (m *manager) syncSVIDs(ctx context.Context) (err error) {
+	// perform syncSVIDs only if using LRU cache
+	if m.c.SVIDCacheMaxSize > 0 {
+		m.cache.SyncSVIDsWithSubscribers()
+		staleEntries := m.cache.GetStaleEntries()
+		if len(staleEntries) > 0 {
+			return m.updateSVIDs(ctx, staleEntries, m.cache)
```

Comment:
Will this contend with the normal sync goroutine? could this cause two goroutines to pick up the same set of stale entries and effectively request the server sign the same SVID twice?

Reply:
Good point. There's a chance that this may happen. Updated code to put fetching of SVIDs under lock.